### PR TITLE
feat: profile forms

### LINKED ISSUE #5575: [FEATURE] Manage profile with a form
### Description

Page to manage profile of authenticated user

- toggle switch to use json file or form
- profile (name, bio, etc)
- links
- social shortcuts
- events
- testimonials

Conditions
- profile form with json file data if new (not existing)
- display alert if json file and MongoDB entry both exist - recommend they remove the json file

### TODO

- [x] manage page
- [x] update basic profile
- [x] form tags
- [x] ~~toggle between file and database source~~ no longer needed
- [x] add manage link to navbar
- [ ] ~~toggle profile enable/disable~~ low priority
- [ ] ~~associate profile to the logged in account~~ moved to another issue https://github.com/EddieHubCommunity/LinkFree/issues/7065
- [ ] ~~if username is different from logged in user, show a warning and have a button to update (this will only work if someone logged in before they changed their username)~~ moved to separate issue https://github.com/EddieHubCommunity/LinkFree/issues/7065
- [x] warning if json file exists
- [x] form links + socials
- [x] form testimonials
- [x] form milestones
- [x] form events

**notes:**
- must be able to handle multiple profiles in the database
- rename the profile (username) if available

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by eddiejaoude**
**NOTE: needs updating to use Mongo from PR** https://github.com/EddieHubCommunity/LinkFree/pull/5765

## Fixes Issue

closes #5575 

To access forms visit `/account/manage/profile`

TODO
- [ ] profile form
- [ ] tags form
- [ ] links form (+ social shortcut toggle)
- [ ] testimonials form
- [ ] events form
- [ ] milestones form
- [ ] main navbar to access profile form and statistics page

Screenshot_2023-02-13_at_22 18 04

## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/6638"></a>

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/6875"></a>

### CODE REVIEW

File: pages/api/account/manage/link.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  const session = await getServerSession(req, res, authOptions);
+  const username = session.username;
+
+  await connectMongo();
+  const log = logger.child({ username });
+
+  let getProfile = await Profile.findOne({ username });
+  if (!getProfile) {
+    return res.status(400).json({ error: "Profile not found" });
```
Comment (GrahamTheDevRel) [inline pages/api/account/manage/link.js:24]: Should it not be a 404 status?

Comment (eddiejaoude) [reply pages/api/account/manage/link.js:24]: Oh yes you are correct, thank you 👍

**Comment by eddiejaoude** [comment]
@iamthecloverly please do not approve pull requests that is still in draft and have a list of tasks still to be completed in the pull request description - this causes confusion in the project and community, you have been warned about this already before

**Comment by iamthecloverly** [comment]
> @iamthecloverly please do not approve pull requests that is still in draft and have a list of tasks still to be completed in the pull request description - this causes confusion in the project and community, you have been warned about this already before

Sorry for that!
Won't repeat again

**Comment by eddiejaoude** [comment]
Closing and raising a new PR https://github.com/EddieHubCommunity/LinkFree/pull/6998

**Pull request closed without merging**